### PR TITLE
DM-15871: Move doImport out of daf_butler.

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: python/lsst/daf/butler/butler.py [hunk 1]
```diff
@@ -27,7 +27,8 @@
 import contextlib
 import logging
 
-from .core.utils import doImport, transactional
+from lsst.utils import doImport
```
Comment (timj) [inline python/lsst/daf/butler/butler.py:30]: You probably should move the lsst.utils imports before the butler imports.

**Review by timj — APPROVED** [review]
Only comment is to fix the order of all the imports so that lsst.utils is above the butler internal imports.

File: python/lsst/daf/butler/butler.py (hunk 1 above)
Comment (TallJimbo) [reply python/lsst/daf/butler/butler.py:30]: It is already, isn't it?

Comment (timj) [reply python/lsst/daf/butler/butler.py:30]: 😄 I think there might be one that isn't later on... Sorry.

**Comment by TallJimbo** [comment]
@timj I've pushed a broader cleanup to the import order and style (since I found some pre-existing things).  Please make sure you don't hate it.

**Review by timj — APPROVED** [review]
Looks good to me.

**Pull request merged**